File: app/src/main/java/com/claude/td/MainActivity.java
package com.claude.td;

import android.app.Activity;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;

/** Single-activity host for the game surface. */
public class MainActivity extends Activity {

    private GameView view;

    @Override
    protected void onCreate(Bundle state) {
        super.onCreate(state);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
        view = new GameView(this);
        setContentView(view);
        immersive();
    }

    @Override
    public void onWindowFocusChanged(boolean hasFocus) {
        super.onWindowFocusChanged(hasFocus);
        if (hasFocus) immersive();
    }

    private void immersive() {
        if (Build.VERSION.SDK_INT < 19) return;
        view.setSystemUiVisibility(
                View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                        | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                        | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                        | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                        | View.SYSTEM_UI_FLAG_FULLSCREEN
                        | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
    }

    @Override
    protected void onPause() {
        super.onPause();
        view.pauseGame();
        view.stop();
    }

    @Override
    protected void onResume() {
        super.onResume();
        view.start();
    }

    @Override
    public void onBackPressed() {
        if (!view.onBack()) super.onBackPressed();
    }
}

File: app/src/main/java/com/multiverse/rift/MainActivity.kt
package com.multiverse.rift

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.view.WindowManager
import com.multiverse.rift.game.GameState
import com.multiverse.rift.render.GameView

/**
 * Одна Activity, один View. Всё остальное рисуется на канве.
 */
class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)

        val prefs = getSharedPreferences(PREFS, Context.MODE_PRIVATE)
        val game = GameState(prefs.getInt(KEY_BEST, 0)) { best ->
            prefs.edit().putInt(KEY_BEST, best).apply()
        }
        setContentView(GameView(this, game))
    }

    private companion object {
        const val PREFS = "rift"
        const val KEY_BEST = "best_depth"
    }
}
